src/systimer_posix.c

```diff
@@ -0,0 +1,102 @@
+/* Could live without sigaction but it seems to be pretty universal */
+
+/* This uses the POSIX 2001 API
+   which allows per-thread timing and minimises risk of
+   interference with other code using timers.
+
+   Sadly it's not always available, so we have an alternative implementation
+   below using the odler setitimer interface */
+
+/* Handler for the Alarm signal */
+
+int SyHaveAlarms = 1;
+
+/* This API lets us pick wich signal to use */
+#define TIMER_SIGNAL SIGVTALRM
+
+
+/* For now anyway we create one timer at initialisation and use it */
+static timer_t syTimer = 0;
+
+#if defined(SYS_IS_CYGWIN32) || defined(SYS_IS_FREEBSD)
+#define MY_CLOCK CLOCK_REALTIME
+#else
+#define MY_CLOCK CLOCK_THREAD_CPUTIME_ID
+#endif
+
```

Comment:
Actually, doesn't the above mean that on FreeBSD and Cygwin, wallclock time is used, but elsewhere CPU time?

That seems like a really bad idea. But perhaps I am missing something?